typescript/sdk/src/core/adapters/SealevelCoreAdapter.ts

```diff
@@ -0,0 +1,149 @@
+import { PublicKey } from '@solana/web3.js';
+
+import { HexString, pollAsync } from '@hyperlane-xyz/utils';
+
+import { BaseSealevelAdapter } from '../../app/MultiProtocolApp';
+import {
+  ProviderType,
+  TypedTransactionReceipt,
+} from '../../providers/ProviderType';
+import { ChainName } from '../../types';
+import { CoreAddresses } from '../contracts';
+
+import { ICoreAdapter } from './types';
+
+// https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/sealevel/programs/mailbox/src/processor.rs
+const MESSAGE_DISPATCH_LOG_REGEX = /Dispatched message to (.*), ID (.*)/;
+
+// This adapter just routes to the HyperlaneCore
+// Which implements the needed functionality for EVM chains
+export class SealevelCoreAdapter
+  extends BaseSealevelAdapter<CoreAddresses>
+  implements ICoreAdapter
+{
+  extractMessageIds(
+    sourceTx: TypedTransactionReceipt,
+  ): Array<{ messageId: HexString; destination: ChainName }> {
+    if (sourceTx.type !== ProviderType.SolanaWeb3) {
+      throw new Error(
+        `Unsupported provider type for SealevelCoreAdapter ${sourceTx.type}`,
+      );
+    }
+    const receipt = sourceTx.receipt;
+    const logs = receipt.meta?.logMessages;
+    if (!logs)
+      throw new Error('Transaction logs required to check message delivery');
+    const parsedLogs = SealevelCoreAdapter.parseMessageDispatchLogs(logs);
+    if (!parsedLogs.length) throw new Error('Message dispatch log not found');
+    return parsedLogs.map(({ destination, messageId }) => ({
+      messageId: Buffer.from(messageId, 'hex').toString('hex'),
```

Comment:
curious why the hex string -> buffer -> hex string conversion is needed?

Reply:
To validate that it is valid hex. I'd rather fail early